tarantool/mesh_connection.py

```diff
@@ -63,3 +131,39 @@ def _opt_reconnect(self):
                 self.port = addr['port']
         else:
             raise NetworkError
+
+        if self.authenticated and self.get_nodes_function_name:
+            now = time.time()
+            if now - self.last_nodes_refresh > self.nodes_refresh_interval:
+                self.refresh_nodes(now)
+
+    def refresh_nodes(self, cur_time):
+        '''
+        Refreshes nodes list by calling Lua function with name
+        self.get_nodes_function_name on the current node. If this field is None
+        no refresh occurs. Usually you don't need to call this function manually
+        since it's called automatically during reconnect every
+        self.nodes_refresh_interval seconds.
+        '''
+        resp = super(MeshConnection, self).call_ex(self.get_nodes_function_name,
+                                                   [], reconnect=False)
+
+        if not (resp.data and resp.data[0]):
+            return
+
+        addrs_raw = resp.data[0]
+        if type(addrs_raw) is list:
+            addrs = []
+            for uri_str in addrs_raw:
+                addr = parse_uri(uri_str)
+                if addr:
+                    addrs.append(addr)
+
+            self.strategy = self.strategy_class(addrs)
```

Comment:
Here I'm in thinking whether it worth to rewind a current instance index to the real one if it is in the new list. At least it will keep index in a strategy class consistent with the real instance in use and possible safe us from possible problems in the future.

Look at the relevant logic in tarantool-java: https://github.com/tarantool/tarantool-java/blob/0ee7b300eae71ac645d8616205e228b25e8e52ad/src/main/java/org/tarantool/RoundRobinSocketProviderImpl.java#L71-L81

Reply:
Refactored

Reply:
As far as I see the current implementation does not adjust a position of the current address too.